addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/handler/UniFiControllerThingHandler.java

```diff
@@ -0,0 +1,359 @@
+
+    private void cancelRefreshJob() {
+        synchronized (this) {
+            if (refreshJob != null) {
+                logger.debug("Cancelling refresh job");
+                refreshJob.cancel(true);
+                refreshJob = null;
+            }
+        }
+    }
+
+    private void run() {
+        try {
+            logger.trace("Executing refresh job");
+            refresh();
+            updateStatus(ONLINE);
+        } catch (UniFiCommunicationException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+        } catch (UniFiInvalidCredentialsException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_CREDENTIALS);
+        } catch (Exception e) {
+            logger.warn("Unhandled exception while refreshing the UniFi Controller {} - {}", getThing().getUID(),
+                    e.getMessage());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void refresh() throws UniFiException {
+        if (controller != null) {
+            logger.debug("Refreshing the UniFi Controller {}", getThing().getUID());
+            // mgb: refresh the controller thing
+            synchronized (this) {
+                sitesCache = getSites();
+                devicesCache = getDevices();
+                clientsCache = getClients();
+                insightsCache = getInsights();
+            }
+            // mgb: then refresh all the client things
+            getThing().getThings().forEach((thing) -> {
+                if (thing.getHandler() instanceof UniFiClientThingHandler) {
+                    ((UniFiClientThingHandler) thing.getHandler()).refresh();
+                }
+            });
+        }
+    }
+
+    private Map<String, UniFiSite> getSites() throws UniFiException {
+        Map<String, UniFiSite> siteMap = new HashMap<>();
+        UniFiSite[] sites = controller.getSites();
+        logger.debug("Found {} UniFi Site(s):", sites.length);
+        for (UniFiSite site : sites) {
+            logger.debug("  {}", site);
+            siteMap.put(site.getId(), site);
+        }
+        return siteMap;
+    }
+
+    private Map<String, UniFiDevice> getDevices() throws UniFiException {
+        Map<String, UniFiDevice> deviceMap = new HashMap<>();
+        Collection<UniFiSite> sites = sitesCache.values();
+        for (UniFiSite site : sites) {
+            Map<String, UniFiDevice> devices = getDevices(site);
+            deviceMap.putAll(devices);
+        }
+        return deviceMap;
+    }
+
+    private Map<String, UniFiDevice> getDevices(UniFiSite site) throws UniFiException {
+        Map<String, UniFiDevice> deviceMap = new HashMap<>();
+        UniFiDevice[] devices = controller.getDevices(site);
+        logger.debug("Found {} UniFi Device(s):", devices.length);
+        for (UniFiDevice device : devices) {
+            device.setSite(sitesCache.get(device.getSiteId()));
```

Comment:
Should this not be the paramter `site` instead of the value from `sitesCache`? I'm assuming it would be the same.

Reply:
Fixed